espnet/nets/scorers/ngram.py

```diff
@@ -78,7 +74,29 @@ def score(self, y, state, x):
                 and next state list for ys.
 
         """
-        return self.score_partial_(y, torch.tensor(range(len(self.chardict))), state, x)
+        return self.score_partial_(y, torch.tensor(range(self.charlen)), state, x)
+
+    def batch_score(self, ys, states, xs):
```

Comment:
This for-loop based implementation looks general. Is it possible to pull this function into `BatchScorerInterface` as default impl?

Reply:
Could you explain it?
I am not sure I totally understand your words

Reply:
I think it is nice to have this for-loop based implementation for all scorers by default when the scorer inherits `BatchScorerInterface`. This is because your implementation requires nothing but the `score` method.

Reply:
I suggest you to replace this default `NotImplementedError`
https://github.com/espnet/espnet/blob/7e6be051bd575bd82310d7d460ab68b7394c7926/espnet/nets/scorer_interface.py#L86-L103
with your new for-loop implementation.

Reply:
yes, good suggestion.
Since that, and I think we can have a test from run.sh through batchscorerinterface right now.

Reply:
but it still needs a variable of 'charlen' which means how much chars in the dict. And suggestion?